aea/agent_loop.py

```diff
@@ -85,16 +102,28 @@ def teardown(self):  # pylint: disable=no-self-use
         # start and stop methods are classmethods cause one instance shared across muiltiple threads
         ExecTimeoutThreadGuard.stop()
 
-    async def run_loop(self) -> None:
+    async def run(self) -> None:
         """Run agent loop."""
         self.logger.debug("agent loop started")
+        self._state.set(AgentLoopStates.starting)
+        self.setup()
```

Comment:
Should `setup` and `teardown` not be private as well?

Reply:
+